grails-app/controllers/org/pih/warehouse/user/UserController.groovy

```diff
@@ -301,7 +301,7 @@ class UserController {
         // If we are in localization mode and want to change the language through the footer on React side
         // we want to make sure we set the language to the clicked one, not for the previous locale
         // we want to set the prev locale only if we disable localization mode through "Disable localization mode" button
-        Locale locale = params?.locale ? LocalizationUtil.getLocale(params.locale) : session?.previousLocale
+        Locale locale = params?.locale ? StringUtils.parseLocaleString(params.locale) : session?.previousLocale
```

Comment:
Now that I'm seeing this used more and more, do we really need to keep the current implementation of LocalizationUtil.getLocale()? It seems like it might be better to just leave all of these LocalizationUtil calls and change the underlying implementation to use StringUtils.parseLocaleString().
```
static Locale getLocale(String localeCode) {
    return StringUtils.parseLocaleString(localeCode)
}
```

You could even add a lenient flag to the method to switch to the more strict parseLocale method which expects a BCP 47 language tag.
 
https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/util/StringUtils.html#parseLocale(java.lang.String)